cmd/minishift/cmd/config/config.go

```diff
@@ -160,7 +160,7 @@ var (
 	DnsmasqContainerImage = createConfigSetting("network-dnsmasq-container", SetString, nil, nil, true, nil)
 
 	// Hyper-V vSwitch set to Default Switch by default
-	HypervVirtualSwitch = createConfigSetting("hyperv-virtual-switch", SetString, []setFn{validations.IsValidHypervVirtualSwitch}, nil, true, "Default Switch")
+	HypervVirtualSwitch = createConfigSetting("hyperv-virtual-switch", SetString, []setFn{validations.IsValidHypervVirtualSwitch}, nil, true, nil)
```

Comment:
This would mean we can NOT define platform/hypervisor specific default values?
Perhaps this can be solved in a different way... as defining it here seems the more obvious choice.

Reply:
We can specify platform/hypervisor  specific default values. But we need to add more if condition to the `func initStartFlags() *flag.FlagSet {`

Reply:
There should be a single way to do this. At the moment it can be done as the config setting, and another way is from the startflags.

It would make more sense if defaults are inside the config section (and only here). Food for thought,

Reply:
```diff
-..., default string)
+..., default string, appliedOnStart bool)
```